java/server/src/org/openqa/grid/common/RegistrationRequest.java

```diff
@@ -280,11 +149,9 @@ public static RegistrationRequest getNewInstance(String json) {
             .convert(DesiredCapabilities.class, capabilities.get(i));
         request.capabilities.add(cap);
       }
-      request.ensureBackwardCompatibility();
       return request;
     } catch (JsonSyntaxException e) {
-      // Check if it was a Selenium Grid 1.0 request.
-      return parseGrid1Request(json);
+      return null;
```

Comment:
parseGrid1Request looks like it either would throw or return RegirstaionRequest. Should this just throw? Or at least return an empty request object?